### PR TITLE
Move choice of panning positions into ladspa module

### PR DESCRIPTION
**Pull request by madwort**
* for 2/3 clients
* first step on the way to making this more similar to the other case?

### CODE REVIEW

File: jacktrip_pypatcher.py [hunk 1]
```diff
@@ -151,15 +151,19 @@ def autopatch(jackClient, dry_run, jacktrip_clients):
...
-        ladspa_mid_right_1 = ladspa.get_port(jackClient, all_panning_positions[6], all_ladspa_ports)
-        ladspa_mid_right_2 = ladspa.get_port(jackClient, 0.46, all_ladspa_ports)
+        ladspa_mid_left_1 = ladspa.get_port(jackClient, panning_positions[0], all_ladspa_ports)
+        ladspa_mid_left_2 = ladspa.get_port(jackClient, panning_positions[1], all_ladspa_ports)
+        ladspa_mid_right_1 = ladspa.get_port(jackClient, panning_positions[2], all_ladspa_ports)
+        ladspa_mid_right_2 = ladspa.get_port(jackClient, panning_positions[3], all_ladspa_ports)
 
     if len(jacktrip_clients) == 2:
 
+        ladspa_positions = ladspa.get_panning_positions(len(jacktrip_clients))
```
Comment (madwort) [inline jacktrip_pypatcher.py:165]: this is a typo!

File: ladspa_plugins.py [hunk 2]
```diff
@@ -7,6 +7,9 @@
 
 
 def get_panning_positions(number_of_clients, all_panning_positions):
+  if number_of_clients == 2 or number_of_clients == 3:
+    return [all_panning_positions[5], (all_panning_positions[5]-0.01),
+      all_panning_positions[6], (all_panning_positions[6]+0.01)]
```
Comment (sandreae) [inline ladspa_plugins.py:12]: Ah, very nice!

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -16,3 +16,4 @@ jobs:
         python -m pip install --upgrade pip
         pip install -r requirements.txt
     - run: python -m black .
+    - run: python -m pytest test_ladspa_plugins.py
```
Comment (madwort) [inline .github/workflows/main.yml:19]: also, run these tests on CI!

File: test_ladspa_plugins.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        0.3,
+        -0.45,
+        0.45,
+        -0.6,
+        0.6,
+        -0.75,
+        0.75,
+    ]
+
+    assert ladspa.get_panning_positions(2, panning_positions) == [-0.45, -0.46, 0.45, 0.46]
```
Comment (madwort) [inline test_ladspa_plugins.py:19]: good to be sure of this...

**Review by sandreae — APPROVED** [review]
Caught one error, apart from that looks good. I'm gunna merge.

File: jacktrip_pypatcher.py [hunk 5]
```diff
@@ -143,7 +143,7 @@ def autopatch(jackClient, dry_run, jacktrip_clients):
 
     if len(jacktrip_clients) == 2 or len(jacktrip_clients) == 3:
 
-        panning_positions = get_panning_positions(len(jacktrip_clients), all_panning_positions)
+        panning_positions = ladspa.get_panning_positions(len(jacktrip_clients), all_panning_positions)
```
Comment (madwort) [inline jacktrip_pypatcher.py:146]: doh!

Comment (madwort) [reply jacktrip_pypatcher.py:146]: (good spot! need automated tests to catch stuff like this!)

**Pull request merged**